### ISSUE TITLE
Stub controller helper method on rspec rails 3

### ISSUE THREAD

**Issue by bolshakov** [rspec/rspec-rails#1076]
Before Rspec 3 I've stubbed controller helper method this way:

```ruby
before do 
   expect(view).to receive(:current_customer).and_return(FactoryGirl.build(:customer)
end
```

Now, i'm getting error: `#<#<Class:0x007fc72addeea8>:0x007fc72add56f0 .... >does not implement: current_customer `

Thats right. There is no method `current_customer` on rendered view. So I can't stub it.

There is no such method on anonymous controller too. 

So how should I stub controller helpers while testing view?

**Comment by nruth** [comment]
I didn't see a drop-in solution posted so I tried config blocks.
This stopped my view tests failing (stub is_admin? on application controller), but I didn't check if it turns verification on again as intended.

```
RSpec.configure do |config|
  config.before(:each, type: :view) do
    config.mock_with :rspec do |mocks|
      mocks.verify_partial_doubles = false
    end
  end

  config.after(:each, type: :view) do
    config.mock_with :rspec do |mocks|
      mocks.verify_partial_doubles = true
    end
  end
end
```

**Comment by vpuzzella** [comment]
I did something similar to what @nruth suggested:
```
  config.around(:each, type: :view) do |ex|
    config.mock_with :rspec do |mocks|
      mocks.verify_partial_doubles = false
      ex.run
      mocks.verify_partial_doubles = true
    end
  end
```

**Comment by samphippen** [comment]
@JonRowe I saw you locked this thread. Do you think we can close this issue? If not, can you provide a summary of the work that was agreed to be done?

**Comment by samphippen** [comment]
Closing as dup of #1076. @JonRowe if you do have comments can you leave them there?

**Comment by JonRowe** [comment]
@samphippen this is #1076 ;) it was only locked due to me too comments at a guess, (I don't remember)

**Comment by samphippen** [comment]
Sorry I meant: #1219 for the dup

**Comment by JonRowe** [comment]
It's not a duplicate

**Comment by samphippen** [comment]
I thought so too at first, but the ask is basically the same. That is, people used to be stubbing locals/helper methods, and now they're not able to. They're both stubbing on views unless I'm mistaken?

**Comment by JonRowe** [comment]
#1219 is similar but not identical, as this refers to broken controller stubs due to controller methods not being present in views, where as #1219 refers to new functionality for inject locals into views.

**Comment by JonRowe** [comment]
The difference being that this is "broken" due to verifying partial doubles, methods which exist on controllers should be stub-able with partial verification turned on, where as #1219 is creating stubs for things that don't exist.

**Comment by samphippen** [comment]
This issue starts with asking to stub a method which did not exist, which is the same thing right?

**Comment by JonRowe** [comment]
No, because the method only doesn't exist here due to our isolation of views from controllers

**Comment by samphippen** [comment]
Ah OK. That makes sense, but was hard to parse out here.

However, I'm inclined to close both issues and open something on mocks which makes it easier to disable verify partial doubles, which sort of cleanly provides a path for both use cases.

Thoughts @JonRowe ?

**Comment by JonRowe** [comment]
That's a neater workaround, but it doesn't "fix" the issue, I'm not all that invested in view specs so this workaround is acceptable to me, after the mocks implementation is done we should automatically do this for all view specs

**Comment by samphippen** [comment]
While it doesn't "fix" the issue, it does mean users can trivially opt in to a work-around, which I feel is probably good enough. Thoughts?

**Comment by samphippen** [comment]
As you mentioned, view specs are an edge case, so I feel like a generic but not totally complete fix is useful is my reasoning.

**Comment by JonRowe** [comment]
👍 I'm ok with probably good enough.

**Comment by samphippen** [comment]
cool.

**Comment by samphippen** [comment]
Closing in favour of https://github.com/rspec/rspec-rails/issues/1076#issuecomment-230463271
